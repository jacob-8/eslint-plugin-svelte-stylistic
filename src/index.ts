import bracketsSameLine from './rules/brackets-same-line'
import consistentAttributeLines from './rules/consistent-attribute-lines'

export default {
  rules: {
    'brackets-same-line': bracketsSameLine,
    'consistent-attribute-lines': consistentAttributeLines,
  },
}
