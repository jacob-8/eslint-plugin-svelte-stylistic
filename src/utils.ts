import { ESLintUtils } from '@typescript-eslint/utils'

const hasDocs: string[] = []

export const createEslintRule = ESLintUtils.RuleCreator(
  ruleName => hasDocs.includes(ruleName)
    ? `https://github.com/jacob-8/eslint-plugin-svelte-stylistic/blob/main/src/rules/${ruleName}.md`
    : `https://github.com/jacob-8/eslint-plugin-svelte-stylistic/blob/main/src/rules/${ruleName}.test.ts`,
)
