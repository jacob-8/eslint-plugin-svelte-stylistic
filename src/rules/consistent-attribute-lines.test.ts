/* eslint-disable ts/quotes */
import { RuleTester } from '../../vendor/rule-tester/src/RuleTester'
import rule, { RULE_NAME } from './consistent-attribute-lines'

const valids: string[] = [
  `<div>Text</div>`,

  `<div />`,

  `<div baz foo bar="2">Text</div>`,

  `<div
    baz
    foo
    bar="2">Text</div>`,

  `<div
    foo
    use:baz={{
      setting: 'value',
    }} />`,
]

// See snapshot for fixed code
const invalids = [
  `<div baz
    foo
    bar="2">Text</div>`,

  `<div baz foo
    bar="2">Text</div>`,

  `<div
    baz foo bar="2">Text</div>`,

  `<div foo
    bar />`,

  `<div foo use:baz={{
              setting: 'value',
            }} />`,

  `<div foo use:baz={{
              setting: 'value',
            }} use:box />`,
]

const ruleTester: RuleTester = new RuleTester({
  parser: require.resolve('svelte-eslint-parser'),
})

ruleTester.run(RULE_NAME, rule as any, {
  valid: valids,
  invalid: invalids.map(i => ({
    code: i,
    errors: null,
    onOutput: (output: string) => {
      expect(output).toMatchSnapshot()
    },
  })),
})
