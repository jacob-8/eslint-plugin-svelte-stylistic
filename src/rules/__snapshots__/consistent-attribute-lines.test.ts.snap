// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`consistent-attribute-lines > invalid > <div
    baz foo bar="2">Text</div> 1`] = `
"<div
    baz
foo
bar=\\"2\\">Text</div>"
`;

exports[`consistent-attribute-lines > invalid > <div baz
    foo
    bar="2">Text</div> 1`] = `"<div baz foo bar=\\"2\\">Text</div>"`;

exports[`consistent-attribute-lines > invalid > <div baz foo
    bar="2">Text</div> 1`] = `"<div baz foo bar=\\"2\\">Text</div>"`;

exports[`consistent-attribute-lines > invalid > <div foo
    bar /> 1`] = `"<div foo bar />"`;

exports[`consistent-attribute-lines > invalid > <div foo use:baz={{
              setting: 'value',
            }} /> 1`] = `
"<div
foo
use:baz={{
              setting: 'value',
            }} />"
`;

exports[`consistent-attribute-lines > invalid > <div foo use:baz={{
              setting: 'value',
            }} use:box /> 1`] = `
"<div
foo
use:baz={{
              setting: 'value',
            }}
use:box />"
`;
